hudi-utilities/src/test/java/org/apache/hudi/utilities/transform/TestSqlFileBasedTransformer.java

```diff
@@ -0,0 +1,165 @@
+
+    // Test if the class throws hoodie IO exception correctly when given a incorrect config.
+    props.setProperty(
+        "hoodie.deltastreamer.transformer.sql.file",
+        UtilitiesTestBase.dfsBasePath + "/non-exist-sql-file.sql");
+    assertThrows(
+        HoodieIOException.class,
+        () -> sqlFileTransformer.apply(jsc, sparkSession, inputDatasetRows, props));
+
+    // Test if the SQL file based transformer works as expected for the invalid SQL statements.
+    props.setProperty(
+        "hoodie.deltastreamer.transformer.sql.file",
+        UtilitiesTestBase.dfsBasePath + "/sql-file-transformer-invalid.sql");
+    assertThrows(
+        ParseException.class,
+        () -> sqlFileTransformer.apply(jsc, sparkSession, inputDatasetRows, props));
+
+    // Test if the SQL file based transformer works as expected for the empty SQL statements.
+    props.setProperty(
+        "hoodie.deltastreamer.transformer.sql.file",
+        UtilitiesTestBase.dfsBasePath + "/sql-file-transformer-empty.sql");
+    Dataset<Row> emptyRow = sqlFileTransformer.apply(jsc, sparkSession, inputDatasetRows, props);
+    String[] actualRows = emptyRow.as(Encoders.STRING()).collectAsList().toArray(new String[0]);
+    String[] expectedRows = emptyDatasetRow.collectAsList().toArray(new String[0]);
+    assertArrayEquals(expectedRows, actualRows);
+
+    // Test if the SQL file based transformer works as expected for the correct input.
+    props.setProperty(
+        "hoodie.deltastreamer.transformer.sql.file",
+        UtilitiesTestBase.dfsBasePath + "/sql-file-transformer.sql");
+    Dataset<Row> transformedRow =
+        sqlFileTransformer.apply(jsc, sparkSession, inputDatasetRows, props);
+
+    // Called distinct() and sort() to match the transformation in this file:
+    // hudi-utilities/src/test/resources/delta-streamer-config/sql-file-transformer.sql
+    expectedRows =
+        inputDatasetRows
+            .distinct()
+            .sort("col1")
+            .as(Encoders.STRING())
+            .collectAsList()
+            .toArray(new String[0]);
+    actualRows = transformedRow.as(Encoders.STRING()).collectAsList().toArray(new String[0]);
+    assertArrayEquals(expectedRows, actualRows);
+  }
+
+  private Dataset<Row> getInputDatasetRows() {
+    // Create few rows with duplicate data.
+    List<Row> list = new ArrayList<>();
+    list.add(RowFactory.create("one"));
+    list.add(RowFactory.create("two"));
+    list.add(RowFactory.create("three"));
+    list.add(RowFactory.create("four"));
+    list.add(RowFactory.create("four"));
+    // Create the schema struct.
+    List<org.apache.spark.sql.types.StructField> listOfStructField = new ArrayList<>();
+    listOfStructField.add(DataTypes.createStructField("col1", DataTypes.StringType, true));
+    StructType structType = DataTypes.createStructType(listOfStructField);
+    SparkSession spark = sparkSession.builder().getOrCreate();
```

Comment:
Can we reuse spark from UtilsTestBase. Bcoz, looks like we are leaking the spark session here w/o cleaning it up after the test is complete.